File: app/src/main/java/info/nightscout/androidaps/plugins/Insulin/InsulinOrefRapidActingPlugin.java
package info.nightscout.androidaps.plugins.Insulin;

import info.nightscout.androidaps.MainApp;
import info.nightscout.androidaps.R;

/**
 * Created by adrian on 14/08/17.
 */

public class InsulinOrefRapidActingPlugin extends InsulinOrefBasePlugin {

    private boolean fragmentEnabled = false;
    private boolean fragmentVisible = false;

    private static InsulinOrefRapidActingPlugin plugin = null;

    public static InsulinOrefRapidActingPlugin getPlugin() {
        if (plugin == null)
            plugin = new InsulinOrefRapidActingPlugin();
        return plugin;
    }

    public static final int PEAK = 75;

    @Override
    public int getId() {
        return OREF_RAPID_ACTING;
    }

    @Override
    public String getName() {
        return MainApp.sResources.getString(R.string.rapid_acting_oref);
    }

    @Override
    public String getFragmentClass() {
        return InsulinFragment.class.getName();
    }

    @Override
    public String getFriendlyName() {
        return MainApp.sResources.getString(R.string.rapid_acting_oref);
    }

    @Override
    public String commentStandardText() {
        return MainApp.sResources.getString(R.string.fastactinginsulincomment);
    }

    @Override
    public boolean isEnabled(int type) {
        return type == INSULIN && fragmentEnabled;
    }

    @Override
    public boolean isVisibleInTabs(int type) {
        return type == INSULIN && fragmentVisible;
    }

    @Override
    public void setFragmentEnabled(int type, boolean fragmentEnabled) {
        if (type == INSULIN) this.fragmentEnabled = fragmentEnabled;
    }

    @Override
    public void setFragmentVisible(int type, boolean fragmentVisible) {
        if (type == INSULIN) this.fragmentVisible = fragmentVisible;
    }

    @Override
    public int getPreferencesId() {
        return -1;
    }

    @Override
    int getPeak() {
        return PEAK;
    }
}

File: app/src/main/java/info/nightscout/androidaps/plugins/insulin/InsulinOrefRapidActingPlugin.kt
package info.nightscout.androidaps.plugins.insulin

import dagger.android.HasAndroidInjector
import info.nightscout.androidaps.R
import info.nightscout.androidaps.interfaces.Config
import info.nightscout.androidaps.interfaces.Insulin
import info.nightscout.androidaps.interfaces.ProfileFunction
import info.nightscout.shared.logging.AAPSLogger
import info.nightscout.androidaps.plugins.bus.RxBus
import info.nightscout.androidaps.utils.resources.ResourceHelper
import org.json.JSONObject
import javax.inject.Inject
import javax.inject.Singleton

/**
 * Created by adrian on 14/08/17.
 */
@Singleton
class InsulinOrefRapidActingPlugin @Inject constructor(
    injector: HasAndroidInjector,
    rh: ResourceHelper,
    profileFunction: ProfileFunction,
    rxBus: RxBus,
    aapsLogger: AAPSLogger,
    config: Config
) : InsulinOrefBasePlugin(injector, rh, profileFunction, rxBus, aapsLogger, config) {

    override val id get(): Insulin.InsulinType = Insulin.InsulinType.OREF_RAPID_ACTING
    override val friendlyName get(): String = rh.gs(R.string.rapid_acting_oref)

    override fun configuration(): JSONObject = JSONObject()
    override fun applyConfiguration(configuration: JSONObject) {}

    override fun commentStandardText(): String = rh.gs(R.string.fastactinginsulincomment)

    override val peak = 75

    init {
        pluginDescription
            .pluginName(R.string.rapid_acting_oref)
            .description(R.string.description_insulin_rapid)
            .setDefault()
            .enableByDefault(true)
    }
}